src/handlers/yarn.js

```diff
@@ -0,0 +1,51 @@
+import ora from 'ora';
+import fetch from 'isomorphic-fetch';
+import { exec, log } from '../util';
+
+let originalVersion;
+
+async function afterAll() {
+  await exec(`yarn add ${originalVersion} --no-lockfile`);
+}
+
+async function getCandidates({ module }) {
+  console.log(`Fetch version data from registry.npmjs.org`);
+
+  const response = await fetch(`https://registry.npmjs.org/${module}`);
+  const data = await response.json();
+
+  return Object.keys(data.versions || {});
+}
+
+async function testCandidate(candidate, attempt, { module }) {
+  if (!originalVersion) {
+    const output = JSON.parse(
+      (await exec(`yarn list --pattern ${module} --depth=0 --json`)).stdout
+        .replace("'", '')
+        .replace('\n', ''),
+    );
+    originalVersion = output.data.trees[0].name;
```

Comment:
Does this have a benefit over reading the version from `package.json`? We could even just backup the entire `package.json` file by copying it to a temp dir, and move it back once we're done.

Reply:
No, I don't think there's any benefit to doing it this way. They both should produce the same result.